src/views/studio/CreateEditChannelView/CreateEditChannelView.tsx

```diff
@@ -46,220 +53,240 @@ type CreateEditChannelViewProps = {
     reset,
     errors,
   } = useForm<Inputs>({
     defaultValues: {
       avatar: { url: null, blob: null },
       cover: { url: null, blob: null },
-      channelName: '',
+      title: '',
       description: '',
-      selectedLanguage: languages[0],
+      language: languages[0].value,
       isPublic: true,
     },
   })
 
   const avatarDialogRef = useRef<ImageCropDialogImperativeHandle>(null)
   const coverDialogRef = useRef<ImageCropDialogImperativeHandle>(null)
 
-  const [avatarImageUrl, setAvatarImageUrl] = useState<string | null>(null)
-  const [coverImageUrl, setCoverImageUrl] = useState<string | null>(null)
-  const [isActionBarActive, setActionBarActive] = useState(false)
+  const [transactionStatus, setTransactionStatus] = useState<ExtrinsicStatus | null>(null)
 
   useEffect(() => {
     if (loading || newChannel || !channel) {
       return
     }
-    const { avatarPhotoUrl, coverPhotoUrl, title, description, isPublic, language } = channel
-    const foundLanguage = languages.find(({ name }) => name === language?.name)
+
+    const {
+      avatarPhotoUrl,
+      avatarPhotoAvailability,
+      avatarPhotoDataObject,
+      coverPhotoUrl,
+      coverPhotoAvailability,
+      coverPhotoDataObject,
+      title,
+      description,
+      isPublic,
+      language,
+    } = channel
+
+    const avatarAssetUrl = createUrlFromAsset(avatarPhotoAvailability, avatarPhotoUrl, avatarPhotoDataObject)
+    const coverAssetUrl = createUrlFromAsset(coverPhotoAvailability, coverPhotoUrl, coverPhotoDataObject)
+
+    const foundLanguage = languages.find(({ value }) => value === language?.iso)
+
     reset({
-      ...getValues(),
-      avatar: { blob: null, url: avatarPhotoUrl },
-      cover: { blob: null, url: coverPhotoUrl },
-      channelName: title || '',
+      avatar: { blob: null, url: avatarAssetUrl },
+      cover: { blob: null, url: coverAssetUrl },
+      title: title || '',
       description: description || '',
-      isPublic,
-      selectedLanguage: foundLanguage || languages[0],
+      isPublic: isPublic ?? false,
+      language: foundLanguage?.value || languages[0].value,
     })
-    if (avatarPhotoUrl) {
-      setAvatarImageUrl(avatarPhotoUrl)
-    }
-    if (coverPhotoUrl) {
-      setCoverImageUrl(coverPhotoUrl)
-    }
-  }, [channel, getValues, loading, newChannel, reset])
+  }, [channel, loading, newChannel, reset])
 
-  const handleSubmit = useHandleSubmit((data) => {
-    console.log(data)
+  const handleSubmit = createSubmitHandler(async (data) => {
+    console.log({ dirtyFields })
+    console.log({ data })
+    setTransactionStatus(ExtrinsicStatus.Unsigned)
   })
-  const handleAvatarConfirm = (blob: Blob, url: string) => {
-    setValue('avatar', { url, blob })
-    setAvatarImageUrl(url)
-    setActionBarActive(true)
-  }
-  const handleCoverConfirm = (blob: Blob, url: string) => {
-    setValue('cover', { url, blob })
-    setCoverImageUrl(url)
-    setActionBarActive(true)
```

Comment:
@mikkio-j FYI: I managed to get rid of this by sticking the dialog inside the controller, but if we went this route you can also do `setValue('cover', { url, blob }, { shouldDirty: true })` and this will properly mark the form as dirty so there wouldn't be a need for separate state like `actionBarActive`